### PR TITLE
new logic for state management  implemented

### PR DESCRIPTION
**Pull request by BeAllAround**

### CODE REVIEW

File: src/components/BasicHead.vue [hunk 1]
```diff
@@ -171,20 +171,10 @@ export default {
...
 
   created () {
-    let features = ['print', 'bookmark', 'collect']
-    for(let feature of features) {
-      this.featuresMenu.push(Object.assign({}, 
-        this.tool[feature], 
-        {title: feature, click: this[this.tool[feature].click]}))
-    }
-    // console.log('featuresMenu:  ', this.featuresMenu)
+    this.$store.state.$refs = this.$refs
```
Comment (rjrodger) [inline src/components/BasicHead.vue:174]: is this needed? - this is extremely dangerous - refs point to complex objects with circular references - we should never need them to be observable

File: src/components/BasicHead.vue [hunk 2]
```diff
@@ -206,14 +196,21 @@ export default {
         this.$forceUpdate()
       }
     },
-    route$: {
+    '$route.name': {
       immediate: true,
       handler (val) {
         let name = this.$route.name
+        if(val) {
+          name = val
```
Comment (rjrodger) [inline src/components/BasicHead.vue:204]: won't line 202 and line 204 give the same value?

File: src/components/BasicHead.vue [hunk 3]
```diff
@@ -244,6 +241,20 @@ export default {
   },
   
   methods: {
+    defaults () {
+      let features = ['print', 'bookmark', 'collect']
```
Comment (rjrodger) [inline src/components/BasicHead.vue:245]: concrete data needs to move to the model

File: src/components/BasicHead.vue [hunk 4]
```diff
@@ -244,6 +241,20 @@ export default {
...
+      this.featuresMenu = []
+      for(let feature of features) {
+        if(this.tool[feature]) {
+          this.featuresMenu.push(Object.assign({}, 
+            this.tool[feature], 
+            {title: feature, click: this[this.tool[feature].click]}))
```
Comment (rjrodger) [inline src/components/BasicHead.vue:251]: replace this approach with the way the add button is implemented - all tool buttons need to work the same way or we will lose maintainability

File: src/components/BasicHead.vue [hunk 5]
```diff
@@ -244,6 +241,20 @@ export default {
...
+        }
+      }
+      if(this.tool.select.active) {
+        this.select = this.tool.select.initial
+      }
+    },
     printMap () {
```
Comment (rjrodger) [inline src/components/BasicHead.vue:258]: we may not be printing a map - just call this `print`

File: src/components/BasicSide.vue [hunk 6]
```diff
@@ -17,7 +17,7 @@
       </div>
 
 
-      <v-btn-toggle style="background-color: rgb(var(--vxg-cb1)) !important;"
+      <v-btn-toggle style="background-color: rgb(var(--vxg-cb1)) !important;"
```
Comment (rjrodger) [inline src/components/BasicSide.vue:20]: this needs to be a CSS class

File: src/components/BasicSide.vue [hunk 7]
```diff
@@ -133,27 +133,49 @@ export default {
       menuViewList.push(menuView)
     }
     this.menuViewList = menuViewList
-    this.menuView = menuViewList[0]
+    console.log("menuViewList", this.menuViewList)
+
+    if(this.$route.name != this.custom.special.view) {
```
Comment (rjrodger) [inline src/components/BasicSide.vue:138]: this logic only handles two views - it needs to handle 1..n views

File: src/components/BasicSide.vue [hunk 8]
```diff
@@ -133,27 +133,49 @@ export default {
...
 
       if(pathname && pathname !== this.$route.name ) {
         this.$router.push(pathname)
       }
-    }
+    },
+    '$route.name': {
+      immediate: true,
+      handler (val) {
+        if(val != this.custom.special.view) {
```
Comment (rjrodger) [inline src/components/BasicSide.vue:171]: also 1..n

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: src/components/BasicHead.vue (hunk 1 above)
Comment (BeAllAround) [reply src/components/BasicHead.vue]: Removed! Found a better way to do this

File: src/components/BasicHead.vue (hunk 2 above)
Comment (BeAllAround) [reply src/components/BasicHead.vue:208]: True - I can never be so sure hence writing unnecessary code!

Comment (BeAllAround) [reply src/components/BasicHead.vue]: Removed!

File: src/components/BasicSide.vue (hunk 6 above)
Comment (BeAllAround) [reply src/components/BasicSide.vue]: Done

File: src/components/BasicHead.vue (hunk 4 above)
Comment (BeAllAround) [reply src/components/BasicHead.vue]: Done

File: src/components/BasicHead.vue (hunk 5 above)
Comment (BeAllAround) [reply src/components/BasicHead.vue]: Done

File: src/components/BasicSide.vue (hunk 7 above)
Comment (BeAllAround) [reply src/components/BasicSide.vue]: Done!

File: src/components/BasicSide.vue (hunk 8 above)
Comment (BeAllAround) [reply src/components/BasicSide.vue]: Done!

File: src/components/BasicHead.vue (hunk 3 above)
Comment (BeAllAround) [reply src/components/BasicHead.vue]: Removed - as it is not needed now

**Pull request merged**